docs/source/links/state.md

```diff
@@ -0,0 +1,609 @@
+    }
+
+    if (data.error) {
+      return { error: data.error };
+    }
+
+    return {
+      loading: false,
+      networkStatus,
+      articles,
+    };
+  },
+})(Articles);
+```
+
+Now that you've seen how easy it is to manage your local state in Apollo Client,
+let's dive deeper into how `apollo-link-state` updates & queries your local
+data: the resolver map.
+
+<h2 id="resolver">Resolver map</h2>
+
+Your resolver map is where all the magic happens to retrieve and update your
+local data in the Apollo cache. The resolver map is an object with resolver
+functions for each GraphQL object type. You can think of a GraphQL query or
+mutation as a tree of function calls for each field. These function calls
+resolve to data or another function call.
+
+The three most important things to keep in mind about resolvers in
+`apollo-link-state` are this:
+
+1. The cache is added to the context for you so you can write & read data from
+   the cache.
+2. The resolver should return an object with a `__typename` property unless
+   you've overridden the `dataIdFromObject` function to not use `__typename` for
+   cache keys. This is necessary for Apollo Client to [normalize the data in the
+   cache](/docs/react/basics/caching.html#normalization) properly.
+3. Resolver functions can return a promise if you need to perform asynchronous
+   side effects.
+
+If any of that sounds confusing, I promise it will be cleared up by the end of
+this section. Keep on reading! 😀
+
+<h3 id="default">Default resolvers</h3>
+
+You don't have to specify resolver functions for every field, however. If the
+return value from the parent object has the same property names as the fields
+requested in the child object, you won't need to specify a resolver. This is
+called a [default resolver](docs/graphql-tools/resolvers.html#Default-resolver).
+
+```js
+const getUser = gql`
+  query {
+    user(id: 1) @client {
+      name {
+        last
+        first
+      }
+    }
+  }
+`;
+```
+
+For this query, you will need to specify a resolver for `Query.user` in your
+resolver map. If `Query.user` returns an object with a name property that
+corresponds to an object with last and first properties, you do not need to
+specify any additional resolvers. GraphQL takes care of that for you!
+
+<h3 id="resolver-signature">Resolver signature</h3>
+
+The signature of a resolver function is the exact same as resolver functions on
+the server built with `graphql-tools`. Let's quickly recap the four parameters
+of a resolver function:
+
+```js
+fieldName: (obj, args, context, info) => result;
+```
+
+1. `obj`: The object containing the result returned from the resolver on the
+   parent field or the `ROOT_QUERY` object in the case of a top-level query or
+   mutation. Don't worry about this one too much for `apollo-link-state`.
+2. `args`: An object containing all of the arguments passed into the field. For
+   example, if you called the field with `networkStatus(isConnected: true)`, the
```

Comment:
This matches the example better if you do `updateNetworkStatus` rather than `networkStatus` (I'm not sure what the argument would mean semantically for a query).